prov/efa/src/rxr/rxr_ep.c

```diff
@@ -1803,60 +1804,72 @@ static inline void rxr_ep_check_peer_backoff_timer(struct rxr_ep *ep)
  * @param[in]	ep		end point
  * @param[in]	cqe_to_process	max number of cq entry to poll and process
  */
-static inline void rdm_ep_poll_ibv_cq(struct rxr_ep *ep,
-				      size_t cqe_to_process)
+static inline void rdm_ep_poll_ibv_cq(struct rxr_ep *ep)
 {
-	struct ibv_wc ibv_wc;
+	bool should_end_poll = false;
 	struct efa_cq *efa_cq;
+	/* Initialize an empty ibv_poll_cq_attr struct for ibv_start_poll.
+	 * EFA expects .comp_mask = 0, or otherwise returns EINVAL.
+	 */
+	struct ibv_poll_cq_attr poll_cq_attr = {.comp_mask = 0};
 	struct efa_av *efa_av;
 	struct efa_ep *efa_ep;
 	struct rxr_pkt_entry *pkt_entry;
 	ssize_t ret;
-	int i, err, prov_errno;
+	int err, prov_errno;
 
 	efa_ep = container_of(ep->rdm_ep, struct efa_ep, util_ep.ep_fid);
 	efa_av = efa_ep->av;
 	efa_cq = container_of(ep->rdm_cq, struct efa_cq, util_cq.cq_fid);
-	for (i = 0; i < cqe_to_process; i++) {
-		ret = ibv_poll_cq(efa_cq->ibv_cq, 1, &ibv_wc);
-
-		if (ret == 0)
-			return;
 
-		if (OFI_UNLIKELY(ret < 0 || ibv_wc.status)) {
-			if (ret < 0) {
-				efa_eq_write_error(&ep->util_ep, -ret, -ret);
-				return;
-			}
+	/* Call ibv_start_poll only once */
+	ret = ibv_start_poll(efa_cq->ibv_cq_ex, &poll_cq_attr);
+	if (!ret)
+		should_end_poll = true;
+
+	do {
+		if (OFI_UNLIKELY(ret != 0)) {
```

Comment:
`!=0` is no necessary.

Reply:
I agree. IIUC `OFI_UNLIKELY` is an optimization trick, i.e. `ret != 0` is unlikely to happen, which saves instructions on the happy path.

Not sure about it's significance though. Will remove it.

Reply:
> Not sure about it's significance though. Will remove it.
I take this back. Without `ret != 0` we will end up with more instructions on the happy path, and the code is not easier to understand. I will keep it for now.

Reply:
Can you elaborate about "more instructions"?

Reply:
Currently, if `ret = 0`, then there is only 1 `if(OFI_UNLIKELY(ret != 0))` on the code path.
`
Suppose we remove `ret != 0`, there will be 2 `if` statements, i.e. `if (ret == ENOENT)` & `if (ret < 0)`.